vortex-scalar/src/scalarvalue/utf8.rs

```diff
@@ -0,0 +1,67 @@
+use vortex_buffer::BufferString;
+use vortex_error::{vortex_err, VortexError, VortexExpect, VortexResult};
+
+use crate::scalarvalue::InnerScalarValue;
+use crate::ScalarValue;
+
+impl<'a> TryFrom<&'a ScalarValue> for String {
```

Comment:
I'm not necessarily a fan of all of these "helper" impls, given it's not that hard to do `value.as_buffer().into_string()` or equivalent.

Don't feel too strongly, it's just a lot of code that's not obviously used anywhere.

Reply:
these function are so you can avoid some footguns with pvalues. You have to give me the T to unwrap pvalue into